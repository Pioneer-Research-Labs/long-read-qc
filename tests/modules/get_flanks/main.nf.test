nextflow_process {

    name "Test Process get_flanks"
    script "modules/get_flanks/main.nf"
    process "get_flanks"

    test("Should get insert and barcode flanking sequences") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.fromList([ tuple([ id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/c.00323.dna'))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot("$launchDir/results/Gateway_EmptyBarcodedLib1_20241217/flanking.gb").match()
        }

    }

}
