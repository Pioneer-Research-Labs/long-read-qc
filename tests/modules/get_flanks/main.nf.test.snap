{
    "Should get insert and barcode flanking sequences": {
        "content": [
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/6ac05898421d6a7e04f53a3a6a3899f3/results/Gateway_EmptyBarcodedLib1_20241217/flanking.gb"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-31T17:24:39.586467"
    }
}